File: Bank-backend/src/main/java/com/ipnet/Bank_backend/services/TypeCompteService.java
package com.ipnet.Bank_backend.services;

import com.ipnet.Bank_backend.entities.Compte;
import com.ipnet.Bank_backend.entities.TypeCompte;

import java.util.List;

public interface TypeCompteService {

    TypeCompte create (TypeCompte typeCompte);
    TypeCompte update (TypeCompte typeCompte, Long id);
    List<TypeCompte>  getAll ();
    void  delete (Long id);
    TypeCompte  findById (Long id);
}

File: vue-crud-app/src/services/TypeCompteService.js
import http from "../http-common";

class TypeCompteService {
    getAll() {
        return http.get("/typeCompte/");
    }

    get(id) {
        return http.get(`/typeCompte/${id}`);
    }

    create(data) {
        return http.post("/typeCompte/", data);
    }

    update(id, data) {
        return http.put(`/typeCompte/${id}`, data);
    }

    delete(id) {
        return http.delete(`/typeCompte/${id}`);
    }



    findById(title) {
        return http.get(`/typeCompte?title=${title}`);
    }
}

export default new TypeCompteService();
